pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -938,10 +938,12 @@ func (p *testPlanner) finalizeUpgradeSteps(
 		// auto-upgrading after resetting a cluster setting, we need to
 		// "manually" run the migrations at this point.
 		if toVersion.AtLeast(tenantSupportsAutoUpgradeVersion) {
-			steps = append(
-				steps,
-				p.concurrently(mixedVersionLabel, p.hooks.MixedVersionSteps(p.currentContext, p.prng))...,
-			)
+			if scheduleHooks {
```

Comment:
To facilitate future debugging of this type of issue, we should probably update `MixedVersionSteps`, `AfterUpgradeFinalizedSteps`, etc., with the enclosing context. E.g., instead of the ambiguous,

```
run "wait and validate" (16)
```

we could say something like,

```
mvt.InMixedVersion--"wait and validate" (16)
```

to denote the specific type of user-hook.

Reply:
Theres actually a debug mode used by the unit tests that will print the stage of each step. Let me look into it sometime this week, sounds like exactly what we want. I'll merge this PR for now to stop any flakes.

Reply:
> Theres actually a debug mode used by the unit tests that will print the stage of each step.

With `prettyPrintInternal(true)`, it prints the enclosing stage (`ss.context.System.Stage`), so it may look like,

```
run "wait and validate" (16) [stage=running-upgrade-migrations,finalizing]
```

but it's still _not_ immediately obvious what user-hook is being run.